test/integration/ads_integration_test.cc

```diff
@@ -889,4 +889,236 @@ TEST_P(AdsClusterFromFileIntegrationTest, BasicTestWidsAdsEndpointLoadedFromFile
                                       {"ads_eds_cluster"}, {}, {}));
 }
 
+namespace {
+static std::string AdsIntegrationConfigWithRdts(const std::string& api_type) {
+  // Note: do not use CONSTRUCT_ON_FIRST_USE here!
+  return fmt::format(R"EOF(
+layered_runtime:
+  layers:
+    - name: foobar
+      rtds_layer:
+        name: ads_rtds_layer
+        rtds_config:
+          ads: {{}}
+dynamic_resources:
+  lds_config:
+    ads: {{}}
+  cds_config:
+    ads: {{}}
+  ads_config:
+    api_type: {}
+    set_node_on_first_message_only: true
+static_resources:
+  clusters:
+    name: dummy_cluster
+    connect_timeout:
+      seconds: 5
+    type: STATIC
+    load_assignment:
+      cluster_name: dummy_cluster
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: 127.0.0.1
+                port_value: 0
+    lb_policy: ROUND_ROBIN
+    http2_protocol_options: {{}}
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: 127.0.0.1
+      port_value: 0
+)EOF",
```

Comment:
Do we need to have a full boiler plate config here vs. having bootstrap config modifiers? All things being equal, the later are preferred, since we have less boiler plate to maintain and it's less fragile.

Reply:
Will address in a bit.

Reply:
cleaned up